// @vercel/sdk/esm/models/updateflagop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as types from "../types/primitives.js";
import { smartUnion } from "../types/smartUnion.js";
export const UpdateFlagCmp = {
    Eq: "eq",
    NotEq: "!eq",
    OneOf: "oneOf",
    NotOneOf: "!oneOf",
    ContainsAllOf: "containsAllOf",
    ContainsAnyOf: "containsAnyOf",
    ContainsNoneOf: "containsNoneOf",
    StartsWith: "startsWith",
    NotStartsWith: "!startsWith",
    EndsWith: "endsWith",
    NotEndsWith: "!endsWith",
    Ex: "ex",
    NotEx: "!ex",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    Regex: "regex",
    NotRegex: "!regex",
    Before: "before",
    After: "after",
};
export const UpdateFlagRhsType = {
    ListInline: "list/inline",
    List: "list",
};
export const UpdateFlagState = {
    Active: "active",
    Archived: "archived",
};
/** @internal */
export const UpdateFlagValue$inboundSchema = smartUnion([types.string(), types.number(), types.boolean()]);
/** @internal */
export const UpdateFlagValue$outboundSchema = smartUnion([z.string(), z.number(), z.boolean()]);
export function updateFlagValueToJSON(updateFlagValue) {
    return JSON.stringify(UpdateFlagValue$outboundSchema.parse(updateFlagValue));
}
export function updateFlagValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagValue' from JSON`);
}
/** @internal */
export const UpdateFlagVariants$inboundSchema = z.object({
    id: types.string(),
    label: types.optional(types.string()),
    description: types.optional(types.string()),
    value: smartUnion([types.string(), types.number(), types.boolean()]),
});
/** @internal */
export const UpdateFlagVariants$outboundSchema = z.object({
    id: z.string(),
    label: z.string().optional(),
    description: z.string().optional(),
    value: smartUnion([z.string(), z.number(), z.boolean()]),
});
export function updateFlagVariantsToJSON(updateFlagVariants) {
    return JSON.stringify(UpdateFlagVariants$outboundSchema.parse(updateFlagVariants));
}
export function updateFlagVariantsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagVariants$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagVariants' from JSON`);
}
/** @internal */
export const UpdateFlagReuse$inboundSchema = z.object({
    active: types.boolean(),
    environment: types.string(),
});
/** @internal */
export const UpdateFlagReuse$outboundSchema = z.object({
    active: z.boolean(),
    environment: z.string(),
});
export function updateFlagReuseToJSON(updateFlagReuse) {
    return JSON.stringify(UpdateFlagReuse$outboundSchema.parse(updateFlagReuse));
}
export function updateFlagReuseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagReuse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagReuse' from JSON`);
}
/** @internal */
export const UpdateFlagTargets$inboundSchema = z.object({
    note: types.optional(types.string()),
    value: types.string(),
});
/** @internal */
export const UpdateFlagTargets$outboundSchema = z.object({
    note: z.string().optional(),
    value: z.string(),
});
export function updateFlagTargetsToJSON(updateFlagTargets) {
    return JSON.stringify(UpdateFlagTargets$outboundSchema.parse(updateFlagTargets));
}
export function updateFlagTargetsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagTargets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagTargets' from JSON`);
}
/** @internal */
export const UpdateFlagPausedOutcome$inboundSchema = z.object({
    type: types.optional(z.any()),
    variantId: types.string(),
});
/** @internal */
export const UpdateFlagPausedOutcome$outboundSchema = z.object({
    type: z.any().optional(),
    variantId: z.string(),
});
export function updateFlagPausedOutcomeToJSON(updateFlagPausedOutcome) {
    return JSON.stringify(UpdateFlagPausedOutcome$outboundSchema.parse(updateFlagPausedOutcome));
}
export function updateFlagPausedOutcomeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagPausedOutcome$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagPausedOutcome' from JSON`);
}
/** @internal */
export const UpdateFlagLhs2$inboundSchema = z.object({
    type: types.optional(z.any()),
    kind: types.string(),
    attribute: types.string(),
});
/** @internal */
export const UpdateFlagLhs2$outboundSchema = z.object({
    type: z.any().optional(),
    kind: z.string(),
    attribute: z.string(),
});
export function updateFlagLhs2ToJSON(updateFlagLhs2) {
    return JSON.stringify(UpdateFlagLhs2$outboundSchema.parse(updateFlagLhs2));
}
export function updateFlagLhs2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagLhs2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagLhs2' from JSON`);
}
/** @internal */
export const UpdateFlagLhs1$inboundSchema = z.object({
    type: types.optional(z.any()),
});
/** @internal */
export const UpdateFlagLhs1$outboundSchema = z.object({
    type: z.any().optional(),
});
export function updateFlagLhs1ToJSON(updateFlagLhs1) {
    return JSON.stringify(UpdateFlagLhs1$outboundSchema.parse(updateFlagLhs1));
}
export function updateFlagLhs1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagLhs1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagLhs1' from JSON`);
}
/** @internal */
export const UpdateFlagLhs$inboundSchema = smartUnion([
    z.lazy(() => UpdateFlagLhs2$inboundSchema),
    z.lazy(() => UpdateFlagLhs1$inboundSchema),
]);
/** @internal */
export const UpdateFlagLhs$outboundSchema = smartUnion([
    z.lazy(() => UpdateFlagLhs2$outboundSchema),
    z.lazy(() => UpdateFlagLhs1$outboundSchema),
]);
export function updateFlagLhsToJSON(updateFlagLhs) {
    return JSON.stringify(UpdateFlagLhs$outboundSchema.parse(updateFlagLhs));
}
export function updateFlagLhsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagLhs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagLhs' from JSON`);
}
/** @internal */
export const UpdateFlagCmp$inboundSchema = z.nativeEnum(UpdateFlagCmp);
/** @internal */
export const UpdateFlagCmp$outboundSchema = UpdateFlagCmp$inboundSchema;
/** @internal */
export const UpdateFlagRhs2$inboundSchema = z.object({
    type: types.optional(z.any()),
    pattern: types.string(),
    flags: types.string(),
});
/** @internal */
export const UpdateFlagRhs2$outboundSchema = z.object({
    type: z.any().optional(),
    pattern: z.string(),
    flags: z.string(),
});
export function updateFlagRhs2ToJSON(updateFlagRhs2) {
    return JSON.stringify(UpdateFlagRhs2$outboundSchema.parse(updateFlagRhs2));
}
export function updateFlagRhs2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRhs2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRhs2' from JSON`);
}
/** @internal */
export const UpdateFlagRhsType$inboundSchema = z.nativeEnum(UpdateFlagRhsType);
/** @internal */
export const UpdateFlagRhsType$outboundSchema = UpdateFlagRhsType$inboundSchema;
/** @internal */
export const UpdateFlagItems2$inboundSchema = z.object({
    label: types.optional(types.string()),
    note: types.optional(types.string()),
    value: types.string(),
});
/** @internal */
export const UpdateFlagItems2$outboundSchema = z.object({
    label: z.string().optional(),
    note: z.string().optional(),
    value: z.string(),
});
export function updateFlagItems2ToJSON(updateFlagItems2) {
    return JSON.stringify(UpdateFlagItems2$outboundSchema.parse(updateFlagItems2));
}
export function updateFlagItems2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagItems2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagItems2' from JSON`);
}
/** @internal */
export const UpdateFlagItems1$inboundSchema = z.object({
    label: types.optional(types.string()),
    note: types.optional(types.string()),
    value: types.number(),
});
/** @internal */
export const UpdateFlagItems1$outboundSchema = z.object({
    label: z.string().optional(),
    note: z.string().optional(),
    value: z.number(),
});
export function updateFlagItems1ToJSON(updateFlagItems1) {
    return JSON.stringify(UpdateFlagItems1$outboundSchema.parse(updateFlagItems1));
}
export function updateFlagItems1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagItems1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagItems1' from JSON`);
}
/** @internal */
export const UpdateFlagRhsItems$inboundSchema = smartUnion([
    z.lazy(() => UpdateFlagItems1$inboundSchema),
    z.lazy(() => UpdateFlagItems2$inboundSchema),
]);
/** @internal */
export const UpdateFlagRhsItems$outboundSchema = smartUnion([
    z.lazy(() => UpdateFlagItems1$outboundSchema),
    z.lazy(() => UpdateFlagItems2$outboundSchema),
]);
export function updateFlagRhsItemsToJSON(updateFlagRhsItems) {
    return JSON.stringify(UpdateFlagRhsItems$outboundSchema.parse(updateFlagRhsItems));
}
export function updateFlagRhsItemsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRhsItems$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRhsItems' from JSON`);
}
/** @internal */
export const UpdateFlagRhs1$inboundSchema = z.object({
    type: UpdateFlagRhsType$inboundSchema,
    items: z.array(smartUnion([
        z.lazy(() => UpdateFlagItems1$inboundSchema),
        z.lazy(() => UpdateFlagItems2$inboundSchema),
    ])),
});
/** @internal */
export const UpdateFlagRhs1$outboundSchema = z.object({
    type: UpdateFlagRhsType$outboundSchema,
    items: z.array(smartUnion([
        z.lazy(() => UpdateFlagItems1$outboundSchema),
        z.lazy(() => UpdateFlagItems2$outboundSchema),
    ])),
});
export function updateFlagRhs1ToJSON(updateFlagRhs1) {
    return JSON.stringify(UpdateFlagRhs1$outboundSchema.parse(updateFlagRhs1));
}
export function updateFlagRhs1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRhs1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRhs1' from JSON`);
}
/** @internal */
export const UpdateFlagRhs$inboundSchema = smartUnion([
    z.lazy(() => UpdateFlagRhs2$inboundSchema),
    z.lazy(() => UpdateFlagRhs1$inboundSchema),
    types.string(),
    types.number(),
    types.boolean(),
]);
/** @internal */
export const UpdateFlagRhs$outboundSchema = smartUnion([
    z.lazy(() => UpdateFlagRhs2$outboundSchema),
    z.lazy(() => UpdateFlagRhs1$outboundSchema),
    z.string(),
    z.number(),
    z.boolean(),
]);
export function updateFlagRhsToJSON(updateFlagRhs) {
    return JSON.stringify(UpdateFlagRhs$outboundSchema.parse(updateFlagRhs));
}
export function updateFlagRhsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRhs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRhs' from JSON`);
}
/** @internal */
export const UpdateFlagConditions$inboundSchema = z.object({
    lhs: smartUnion([
        z.lazy(() => UpdateFlagLhs2$inboundSchema),
        z.lazy(() => UpdateFlagLhs1$inboundSchema),
    ]),
    cmp: UpdateFlagCmp$inboundSchema,
    rhs: types.optional(smartUnion([
        z.lazy(() => UpdateFlagRhs2$inboundSchema),
        z.lazy(() => UpdateFlagRhs1$inboundSchema),
        types.string(),
        types.number(),
        types.boolean(),
    ])),
});
/** @internal */
export const UpdateFlagConditions$outboundSchema = z.object({
    lhs: smartUnion([
        z.lazy(() => UpdateFlagLhs2$outboundSchema),
        z.lazy(() => UpdateFlagLhs1$outboundSchema),
    ]),
    cmp: UpdateFlagCmp$outboundSchema,
    rhs: smartUnion([
        z.lazy(() => UpdateFlagRhs2$outboundSchema),
        z.lazy(() => UpdateFlagRhs1$outboundSchema),
        z.string(),
        z.number(),
        z.boolean(),
    ]).optional(),
});
export function updateFlagConditionsToJSON(updateFlagConditions) {
    return JSON.stringify(UpdateFlagConditions$outboundSchema.parse(updateFlagConditions));
}
export function updateFlagConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagConditions' from JSON`);
}
/** @internal */
export const UpdateFlagOutcomeBase$inboundSchema = z.object({
    type: types.optional(z.any()),
    kind: types.string(),
    attribute: types.string(),
});
/** @internal */
export const UpdateFlagOutcomeBase$outboundSchema = z.object({
    type: z.any().optional(),
    kind: z.string(),
    attribute: z.string(),
});
export function updateFlagOutcomeBaseToJSON(updateFlagOutcomeBase) {
    return JSON.stringify(UpdateFlagOutcomeBase$outboundSchema.parse(updateFlagOutcomeBase));
}
export function updateFlagOutcomeBaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagOutcomeBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagOutcomeBase' from JSON`);
}
/** @internal */
export const UpdateFlagOutcome2$inboundSchema = z.object({
    type: types.optional(z.any()),
    base: z.lazy(() => UpdateFlagOutcomeBase$inboundSchema),
    weights: z.record(types.number()),
    defaultVariantId: types.string(),
});
/** @internal */
export const UpdateFlagOutcome2$outboundSchema = z.object({
    type: z.any().optional(),
    base: z.lazy(() => UpdateFlagOutcomeBase$outboundSchema),
    weights: z.record(z.number()),
    defaultVariantId: z.string(),
});
export function updateFlagOutcome2ToJSON(updateFlagOutcome2) {
    return JSON.stringify(UpdateFlagOutcome2$outboundSchema.parse(updateFlagOutcome2));
}
export function updateFlagOutcome2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagOutcome2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagOutcome2' from JSON`);
}
/** @internal */
export const UpdateFlagOutcome1$inboundSchema = z.object({
    type: types.optional(z.any()),
    variantId: types.string(),
});
/** @internal */
export const UpdateFlagOutcome1$outboundSchema = z.object({
    type: z.any().optional(),
    variantId: z.string(),
});
export function updateFlagOutcome1ToJSON(updateFlagOutcome1) {
    return JSON.stringify(UpdateFlagOutcome1$outboundSchema.parse(updateFlagOutcome1));
}
export function updateFlagOutcome1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagOutcome1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagOutcome1' from JSON`);
}
/** @internal */
export const UpdateFlagOutcome$inboundSchema = smartUnion([
    z.lazy(() => UpdateFlagOutcome2$inboundSchema),
    z.lazy(() => UpdateFlagOutcome1$inboundSchema),
]);
/** @internal */
export const UpdateFlagOutcome$outboundSchema = smartUnion([
    z.lazy(() => UpdateFlagOutcome2$outboundSchema),
    z.lazy(() => UpdateFlagOutcome1$outboundSchema),
]);
export function updateFlagOutcomeToJSON(updateFlagOutcome) {
    return JSON.stringify(UpdateFlagOutcome$outboundSchema.parse(updateFlagOutcome));
}
export function updateFlagOutcomeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagOutcome$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagOutcome' from JSON`);
}
/** @internal */
export const UpdateFlagRules$inboundSchema = z.object({
    id: types.string(),
    conditions: z.array(z.lazy(() => UpdateFlagConditions$inboundSchema)),
    outcome: smartUnion([
        z.lazy(() => UpdateFlagOutcome2$inboundSchema),
        z.lazy(() => UpdateFlagOutcome1$inboundSchema),
    ]),
});
/** @internal */
export const UpdateFlagRules$outboundSchema = z.object({
    id: z.string(),
    conditions: z.array(z.lazy(() => UpdateFlagConditions$outboundSchema)),
    outcome: smartUnion([
        z.lazy(() => UpdateFlagOutcome2$outboundSchema),
        z.lazy(() => UpdateFlagOutcome1$outboundSchema),
    ]),
});
export function updateFlagRulesToJSON(updateFlagRules) {
    return JSON.stringify(UpdateFlagRules$outboundSchema.parse(updateFlagRules));
}
export function updateFlagRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRules' from JSON`);
}
/** @internal */
export const UpdateFlagFallthroughBase$inboundSchema = z.object({
    type: types.optional(z.any()),
    kind: types.string(),
    attribute: types.string(),
});
/** @internal */
export const UpdateFlagFallthroughBase$outboundSchema = z.object({
    type: z.any().optional(),
    kind: z.string(),
    attribute: z.string(),
});
export function updateFlagFallthroughBaseToJSON(updateFlagFallthroughBase) {
    return JSON.stringify(UpdateFlagFallthroughBase$outboundSchema.parse(updateFlagFallthroughBase));
}
export function updateFlagFallthroughBaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagFallthroughBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagFallthroughBase' from JSON`);
}
/** @internal */
export const UpdateFlagFallthrough2$inboundSchema = z.object({
    type: types.optional(z.any()),
    base: z.lazy(() => UpdateFlagFallthroughBase$inboundSchema),
    weights: z.record(types.number()),
    defaultVariantId: types.string(),
});
/** @internal */
export const UpdateFlagFallthrough2$outboundSchema = z.object({
    type: z.any().optional(),
    base: z.lazy(() => UpdateFlagFallthroughBase$outboundSchema),
    weights: z.record(z.number()),
    defaultVariantId: z.string(),
});
export function updateFlagFallthrough2ToJSON(updateFlagFallthrough2) {
    return JSON.stringify(UpdateFlagFallthrough2$outboundSchema.parse(updateFlagFallthrough2));
}
export function updateFlagFallthrough2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagFallthrough2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagFallthrough2' from JSON`);
}
/** @internal */
export const UpdateFlagFallthrough1$inboundSchema = z.object({
    type: types.optional(z.any()),
    variantId: types.string(),
});
/** @internal */
export const UpdateFlagFallthrough1$outboundSchema = z.object({
    type: z.any().optional(),
    variantId: z.string(),
});
export function updateFlagFallthrough1ToJSON(updateFlagFallthrough1) {
    return JSON.stringify(UpdateFlagFallthrough1$outboundSchema.parse(updateFlagFallthrough1));
}
export function updateFlagFallthrough1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagFallthrough1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagFallthrough1' from JSON`);
}
/** @internal */
export const UpdateFlagFallthrough$inboundSchema = smartUnion([
    z.lazy(() => UpdateFlagFallthrough2$inboundSchema),
    z.lazy(() => UpdateFlagFallthrough1$inboundSchema),
]);
/** @internal */
export const UpdateFlagFallthrough$outboundSchema = smartUnion([
    z.lazy(() => UpdateFlagFallthrough2$outboundSchema),
    z.lazy(() => UpdateFlagFallthrough1$outboundSchema),
]);
export function updateFlagFallthroughToJSON(updateFlagFallthrough) {
    return JSON.stringify(UpdateFlagFallthrough$outboundSchema.parse(updateFlagFallthrough));
}
export function updateFlagFallthroughFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagFallthrough$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagFallthrough' from JSON`);
}
/** @internal */
export const UpdateFlagEnvironments$inboundSchema = z.object({
    active: types.boolean(),
    reuse: types.optional(z.lazy(() => UpdateFlagReuse$inboundSchema)),
    targets: types.optional(z.record(z.record(z.record(z.array(z.lazy(() => UpdateFlagTargets$inboundSchema)))))),
    pausedOutcome: z.lazy(() => UpdateFlagPausedOutcome$inboundSchema),
    rules: z.array(z.lazy(() => UpdateFlagRules$inboundSchema)),
    fallthrough: smartUnion([
        z.lazy(() => UpdateFlagFallthrough2$inboundSchema),
        z.lazy(() => UpdateFlagFallthrough1$inboundSchema),
    ]),
    revision: types.optional(types.number()),
});
/** @internal */
export const UpdateFlagEnvironments$outboundSchema = z.object({
    active: z.boolean(),
    reuse: z.lazy(() => UpdateFlagReuse$outboundSchema).optional(),
    targets: z.record(z.record(z.record(z.array(z.lazy(() => UpdateFlagTargets$outboundSchema))))).optional(),
    pausedOutcome: z.lazy(() => UpdateFlagPausedOutcome$outboundSchema),
    rules: z.array(z.lazy(() => UpdateFlagRules$outboundSchema)),
    fallthrough: smartUnion([
        z.lazy(() => UpdateFlagFallthrough2$outboundSchema),
        z.lazy(() => UpdateFlagFallthrough1$outboundSchema),
    ]),
    revision: z.number().optional(),
});
export function updateFlagEnvironmentsToJSON(updateFlagEnvironments) {
    return JSON.stringify(UpdateFlagEnvironments$outboundSchema.parse(updateFlagEnvironments));
}
export function updateFlagEnvironmentsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagEnvironments$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagEnvironments' from JSON`);
}
/** @internal */
export const UpdateFlagState$inboundSchema = z.nativeEnum(UpdateFlagState);
/** @internal */
export const UpdateFlagState$outboundSchema = UpdateFlagState$inboundSchema;
/** @internal */
export const UpdateFlagRequestBody$inboundSchema = z.object({
    createdBy: types.optional(types.string()),
    message: types.optional(types.string()),
    variants: types.optional(z.array(z.lazy(() => UpdateFlagVariants$inboundSchema))),
    environments: types.optional(z.record(z.lazy(() => UpdateFlagEnvironments$inboundSchema))),
    seed: types.optional(types.number()),
    description: types.optional(types.string()),
    state: types.optional(UpdateFlagState$inboundSchema),
});
/** @internal */
export const UpdateFlagRequestBody$outboundSchema = z.object({
    createdBy: z.string().optional(),
    message: z.string().optional(),
    variants: z.array(z.lazy(() => UpdateFlagVariants$outboundSchema)).optional(),
    environments: z.record(z.lazy(() => UpdateFlagEnvironments$outboundSchema))
        .optional(),
    seed: z.number().optional(),
    description: z.string().optional(),
    state: UpdateFlagState$outboundSchema.optional(),
});
export function updateFlagRequestBodyToJSON(updateFlagRequestBody) {
    return JSON.stringify(UpdateFlagRequestBody$outboundSchema.parse(updateFlagRequestBody));
}
export function updateFlagRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRequestBody' from JSON`);
}
/** @internal */
export const UpdateFlagRequest$inboundSchema = z.object({
    projectIdOrName: types.string(),
    flagIdOrSlug: types.string(),
    ifMatch: types.optional(types.string()),
    withMetadata: types.optional(types.boolean()),
    teamId: types.optional(types.string()),
    slug: types.optional(types.string()),
    RequestBody: types.optional(z.lazy(() => UpdateFlagRequestBody$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateFlagRequest$outboundSchema = z.object({
    projectIdOrName: z.string(),
    flagIdOrSlug: z.string(),
    ifMatch: z.string().optional(),
    withMetadata: z.boolean().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateFlagRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateFlagRequestToJSON(updateFlagRequest) {
    return JSON.stringify(UpdateFlagRequest$outboundSchema.parse(updateFlagRequest));
}
export function updateFlagRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagRequest' from JSON`);
}
