// @vercel/sdk/esm/models/updateflagsettingsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import * as types from "../types/primitives.js";
export const UpdateFlagSettingsTypeName = {
    Settings: "settings",
};
/** @internal */
export const Labels$inboundSchema = z
    .object({
    label: types.string(),
    value: types.string(),
});
/** @internal */
export const Labels$outboundSchema = z.object({
    label: z.string(),
    value: z.string(),
});
export function labelsToJSON(labels) {
    return JSON.stringify(Labels$outboundSchema.parse(labels));
}
export function labelsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Labels$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Labels' from JSON`);
}
/** @internal */
export const Attributes$inboundSchema = z.object({
    key: types.string(),
    type: types.string(),
    labels: types.optional(z.array(z.lazy(() => Labels$inboundSchema))),
});
/** @internal */
export const Attributes$outboundSchema = z.object({
    key: z.string(),
    type: z.string(),
    labels: z.array(z.lazy(() => Labels$outboundSchema)).optional(),
});
export function attributesToJSON(attributes) {
    return JSON.stringify(Attributes$outboundSchema.parse(attributes));
}
export function attributesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Attributes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Attributes' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsEntities$inboundSchema = z.object({
    kind: types.string(),
    label: types.string(),
    attributes: z.array(z.lazy(() => Attributes$inboundSchema)),
});
/** @internal */
export const UpdateFlagSettingsEntities$outboundSchema = z.object({
    kind: z.string(),
    label: z.string(),
    attributes: z.array(z.lazy(() => Attributes$outboundSchema)),
});
export function updateFlagSettingsEntitiesToJSON(updateFlagSettingsEntities) {
    return JSON.stringify(UpdateFlagSettingsEntities$outboundSchema.parse(updateFlagSettingsEntities));
}
export function updateFlagSettingsEntitiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsEntities$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsEntities' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsRequestBody$inboundSchema = z.object({
    enabled: types.optional(types.boolean()),
    entities: types.optional(z.array(z.lazy(() => UpdateFlagSettingsEntities$inboundSchema))),
    environments: types.optional(z.array(types.string())),
});
/** @internal */
export const UpdateFlagSettingsRequestBody$outboundSchema = z.object({
    enabled: z.boolean().optional(),
    entities: z.array(z.lazy(() => UpdateFlagSettingsEntities$outboundSchema))
        .optional(),
    environments: z.array(z.string()).optional(),
});
export function updateFlagSettingsRequestBodyToJSON(updateFlagSettingsRequestBody) {
    return JSON.stringify(UpdateFlagSettingsRequestBody$outboundSchema.parse(updateFlagSettingsRequestBody));
}
export function updateFlagSettingsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsRequestBody' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsRequest$inboundSchema = z.object({
    projectIdOrName: types.string(),
    teamId: types.optional(types.string()),
    slug: types.optional(types.string()),
    RequestBody: types.optional(z.lazy(() => UpdateFlagSettingsRequestBody$inboundSchema)),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateFlagSettingsRequest$outboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateFlagSettingsRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateFlagSettingsRequestToJSON(updateFlagSettingsRequest) {
    return JSON.stringify(UpdateFlagSettingsRequest$outboundSchema.parse(updateFlagSettingsRequest));
}
export function updateFlagSettingsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsRequest' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsTypeName$inboundSchema = z.nativeEnum(UpdateFlagSettingsTypeName);
/** @internal */
export const UpdateFlagSettingsTypeName$outboundSchema = UpdateFlagSettingsTypeName$inboundSchema;
/** @internal */
export const UpdateFlagSettingsConnections$inboundSchema = z.object({
    edgeConfigId: types.string(),
    edgeConfigItemKey: types.string(),
});
/** @internal */
export const UpdateFlagSettingsConnections$outboundSchema = z.object({
    edgeConfigId: z.string(),
    edgeConfigItemKey: z.string(),
});
export function updateFlagSettingsConnectionsToJSON(updateFlagSettingsConnections) {
    return JSON.stringify(UpdateFlagSettingsConnections$outboundSchema.parse(updateFlagSettingsConnections));
}
export function updateFlagSettingsConnectionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsConnections$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsConnections' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsLabels$inboundSchema = z.object({
    label: types.string(),
    value: types.string(),
});
/** @internal */
export const UpdateFlagSettingsLabels$outboundSchema = z.object({
    label: z.string(),
    value: z.string(),
});
export function updateFlagSettingsLabelsToJSON(updateFlagSettingsLabels) {
    return JSON.stringify(UpdateFlagSettingsLabels$outboundSchema.parse(updateFlagSettingsLabels));
}
export function updateFlagSettingsLabelsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsLabels$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsLabels' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsAttributes$inboundSchema = z.object({
    key: types.string(),
    type: types.string(),
    labels: types.optional(z.array(z.lazy(() => UpdateFlagSettingsLabels$inboundSchema))),
});
/** @internal */
export const UpdateFlagSettingsAttributes$outboundSchema = z.object({
    key: z.string(),
    type: z.string(),
    labels: z.array(z.lazy(() => UpdateFlagSettingsLabels$outboundSchema))
        .optional(),
});
export function updateFlagSettingsAttributesToJSON(updateFlagSettingsAttributes) {
    return JSON.stringify(UpdateFlagSettingsAttributes$outboundSchema.parse(updateFlagSettingsAttributes));
}
export function updateFlagSettingsAttributesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsAttributes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsAttributes' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsFeatureFlagsEntities$inboundSchema = z.object({
    kind: types.string(),
    label: types.string(),
    attributes: z.array(z.lazy(() => UpdateFlagSettingsAttributes$inboundSchema)),
});
/** @internal */
export const UpdateFlagSettingsFeatureFlagsEntities$outboundSchema = z.object({
    kind: z.string(),
    label: z.string(),
    attributes: z.array(z.lazy(() => UpdateFlagSettingsAttributes$outboundSchema)),
});
export function updateFlagSettingsFeatureFlagsEntitiesToJSON(updateFlagSettingsFeatureFlagsEntities) {
    return JSON.stringify(UpdateFlagSettingsFeatureFlagsEntities$outboundSchema.parse(updateFlagSettingsFeatureFlagsEntities));
}
export function updateFlagSettingsFeatureFlagsEntitiesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsFeatureFlagsEntities$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsFeatureFlagsEntities' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsMetadata$inboundSchema = z.object({
    activeFlagCount: types.number(),
    archivedFlagCount: types.number(),
    segmentCount: types.number(),
    packSizeInBytes: types.number(),
    packRevision: types.optional(types.number()),
    configUpdatedAt: types.optional(types.number()),
});
/** @internal */
export const UpdateFlagSettingsMetadata$outboundSchema = z.object({
    activeFlagCount: z.number(),
    archivedFlagCount: z.number(),
    segmentCount: z.number(),
    packSizeInBytes: z.number(),
    packRevision: z.number().optional(),
    configUpdatedAt: z.number().optional(),
});
export function updateFlagSettingsMetadataToJSON(updateFlagSettingsMetadata) {
    return JSON.stringify(UpdateFlagSettingsMetadata$outboundSchema.parse(updateFlagSettingsMetadata));
}
export function updateFlagSettingsMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsMetadata' from JSON`);
}
/** @internal */
export const UpdateFlagSettingsResponseBody$inboundSchema = z.object({
    typeName: UpdateFlagSettingsTypeName$inboundSchema,
    projectId: types.string(),
    ownerId: types.optional(types.string()),
    enabled: types.boolean(),
    environments: z.array(types.string()),
    connections: types.optional(z.array(z.lazy(() => UpdateFlagSettingsConnections$inboundSchema))),
    entities: z.array(z.lazy(() => UpdateFlagSettingsFeatureFlagsEntities$inboundSchema)),
    createdAt: types.optional(types.number()),
    updatedAt: types.optional(types.number()),
    metadata: z.lazy(() => UpdateFlagSettingsMetadata$inboundSchema),
});
/** @internal */
export const UpdateFlagSettingsResponseBody$outboundSchema = z.object({
    typeName: UpdateFlagSettingsTypeName$outboundSchema,
    projectId: z.string(),
    ownerId: z.string().optional(),
    enabled: z.boolean(),
    environments: z.array(z.string()),
    connections: z.array(z.lazy(() => UpdateFlagSettingsConnections$outboundSchema)).optional(),
    entities: z.array(z.lazy(() => UpdateFlagSettingsFeatureFlagsEntities$outboundSchema)),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    metadata: z.lazy(() => UpdateFlagSettingsMetadata$outboundSchema),
});
export function updateFlagSettingsResponseBodyToJSON(updateFlagSettingsResponseBody) {
    return JSON.stringify(UpdateFlagSettingsResponseBody$outboundSchema.parse(updateFlagSettingsResponseBody));
}
export function updateFlagSettingsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateFlagSettingsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateFlagSettingsResponseBody' from JSON`);
}
